activejob/lib/active_job/test_helper.rb

```diff
@@ -678,7 +678,15 @@ def prepare_args_for_assertion(args)
       def round_time_arguments(argument)
         case argument
         when Time, ActiveSupport::TimeWithZone, DateTime
-          argument.change(usec: 0)
+          nsec =
+            if ActiveJob::Serializers.time_precision > 0
+              digits = argument.nsec.to_s.rjust(9, "0")
+              digits[0..ActiveJob::Serializers.time_precision - 1].ljust(9, "0")
+            else
+              0
+            end
+
+          argument.change(nsec: nsec.to_i)
```

Comment:
```suggestion
          nsec = argument.nsec
          nsec -= nsec % (10 ** (9 - ActiveJob::Serializers.time_precision))
          argument.change(nsec: nsec)
```

Reply:
@jonathanhefner Thank you! This works too and is much simplier.